### PR TITLE
Access authorization timeout

### PR DESCRIPTION
**Pull request by kozlice**
As discussed in [related issue](https://github.com/ampproject/amphtml/issues/4344): provide possibility to set custom timeout, but limit value to be less-or-equal 3000 ms in production.

### CODE REVIEW

File: extensions/amp-access/0.1/test/test-amp-access-client.js [hunk 1]
```diff
@@ -53,13 +53,35 @@ describe('AccessClientAdapter', () => {
...
     });
 
+    it('should set authorization timeout if provided', () => {
+      let adapter;
+
+      validConfig['authorizationTimeout'] = 5000;
+      adapter = new AccessClientAdapter(window, validConfig, context);
+      expect(adapter.authorizationTimeout_).to.equal(5000);
+
+      validConfig['authorizationTimeout'] = 1000;
```
Comment (kozlice) [inline extensions/amp-access/0.1/test/test-amp-access-client.js]: I wanted to stub `getMode` here to verify that in production 1s is 1s, and 5s is cut to 3s, but couldn't wrap a single function with Sinon.

Comment (dvoytenko) [inline extensions/amp-access/0.1/test/test-amp-access-client.js]: I usually have success my doing this:
```
import * as mode from './mode';
...
sandbox.stub(mode, 'getMode', ...);
```

**Comment by dvoytenko** [comment]
Thanks a lot! It looks good. One recommendation re:mode stubbing, but otherwise i can merge.

**Comment by kozlice** [comment]
@dvoytenko Thanks for the tip, added test cases for production env.

**Comment by dvoytenko** [comment]
@kozlice Thanks again! LGTM.

**Pull request merged**